annotation_pipeline/utils.py

```diff
@@ -70,20 +71,28 @@ def init_pywren_stats(filename='stats.csv'):
         csvfile.write('Function name,Actions number,Actions memory,Average runtime' + '\n')
 
 
-def append_pywren_stats(func_name, runtime_memory, futures, filename='stats.csv'):
+def append_pywren_stats(futures, runtime_memory, filename='stats.csv'):
     if not os.path.isfile(filename):
         return
 
     if type(futures) != list:
         futures = [futures]
 
     actions_num = len(futures)
-    # actions_mem = map_futures[0].run_status['runtime_memory']
-    actions_mem = runtime_memory
+    func_name = futures[0].invoke_status['func_name']
     average_runtime = np.average([future.run_status['exec_time'] for future in futures])
 
     with open(filename, 'a') as csvfile:
-        csvfile.write(f'{func_name},{actions_num},{actions_mem},{average_runtime}\n')
+        csvfile.write(f'{func_name},{actions_num},{runtime_memory},{average_runtime}\n')
+
+
+def display_pywren_stats(filename='stats.csv'):
+    stats = pd.read_csv(filename)
+    print(stats, '\n')
+
+    unit_price_in_dollars = 0.000017
+    calc_func = lambda row: row[1] * (row[2]/1024) * row[3] * unit_price_in_dollars
+    print('Total PyWren cost:', np.sum(np.apply_along_axis(calc_func, 1, stats)), '$')
```

Comment:
Should COS storage be included? Or is it too small to care about?

Reply:
@gilv any info about COS pricing?

Reply:
@LachlanStuart @omerb01 in general you are right...but currently we are compute bounded..your COS usage is so low that costs is 0 even for your largest dataset. We focus currently to estimate the price of cloud functions